### ISSUE TITLE
Kubernetes fails on docker 1.11

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rancher#4428]
Rancher Version:
1.0.0
Docker Version:
1.11.0

Steps to Reproduce: 
start kubernetes on docker version 1.11, the kubelet docker container won't start:
```
4/15/2016 6:22:12 PMtime="2016-04-15T16:22:12Z" level=fatal msg="open /host/var/run/docker/execdriver/native: no such file or directory"
```

**Comment by alena1108** [comment]
@galal-hussein I don't think k8s yet has support for docker 1.11. The documentation says:

"Docker v1.9.1 is officially recommended. Docker v1.8.3 and Docker v1.10 are supported":

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md/#v121

@deniseschannon we should update Rancher docs with this info as well.

@ibuildthecloud should we update our k8s code base to k8s 1.2.2? There was a bunch of changes to k8s dockertools since 1.2.0

**Comment by thaJeztah** [comment]
They told me they're using this issue for tracking progress on compatibility with docker 1.11; https://github.com/kubernetes/kubernetes/issues/23397

**Comment by galal-hussein** [comment]
@AlmaasAre Yes it worked by downgrading Docker.

**Comment by seeekr** [comment]
I'm a total newbie here wrt both k8s and rancher, and one thing I just ran into is that I tried to give rancher + k8s a spin on the brand new (or relatively new) Docker for Mac beta. As far as I can tell I don't have a way of downgrading Docker to 1.10 or lower with Docker for Mac, and rancher doesn't let me know that I'm about to attempt something that in all likelihood will not work out at all. A failing kubelet container's log messages then lead me to this thread which speaks about compatibility between k8s and docker versions.
It would be fantastic if there was a mention at least in the docs, or even better in the rancher UI, about docker version requirements for running a k8s environment!
(And even better would it be, of course, if I could use the shiny and awesome Docker for Mac with rancher! But getting the version requirements and upgrade spaghetti figured out so this can happen is probably a much bigger task than a notice in the docs or UI :) )

**Comment by gauravve** [comment]
I am having a similar issue with Docker 1.10.3. I tried to setup a k8s environment in GCE using rancher. 

I also added the host using Add hosts using Other option and then choosing Google driver. The host was successfully created. 

![image](https://cloud.githubusercontent.com/assets/2516564/15451624/9800348e-2010-11e6-9f49-60f71333e99b.png)

the kubelet service never starts 

![image](https://cloud.githubusercontent.com/assets/2516564/15451631/c79ee97e-2010-11e6-8c5a-944a156cd556.png)
